wireapp/kalium
# fix: upload big assets OutOfMemory exception (AR-2187)

----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Currently, the app is crashing by out of memory exception, when we are dealing with big files, ie: > 60 MB
This happens because we were handling `ByteArray` (upload and encryption) being an expensive DS for the device to handle.

### Solutions

Change the approach to read the `ByteArray` by chunks and write to a temp file and/or channel

### Testing

#### How to Test

Upload a big file, bigger than 65 MB, the file should be sent to the other client, and when downloading, we should be able to see the file with no corruption, full integrity.

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
